roofit/roofitcore/src/RooMinimizer.cxx

```diff
@@ -692,261 +672,296 @@ RooFitResult* RooMinimizer::save(const char* userName, const char* userTitle)
+   res->setEDM(_theFitter->Result().Edm());
+   res->setCovQual(_theFitter->GetMinimizer()->CovMatrixStatus());
+   res->setStatus(_theFitter->Result().Status());
+   std::vector<double> globalCC;
+   TMatrixDSym corrs(_theFitter->Result().Parameters().size());
+   TMatrixDSym covs(_theFitter->Result().Parameters().size());
+   for (unsigned int ic = 0; ic < _theFitter->Result().Parameters().size(); ic++) {
+      globalCC.push_back(_theFitter->Result().GlobalCC(ic));
+      for (unsigned int ii = 0; ii < _theFitter->Result().Parameters().size(); ii++) {
+         corrs(ic, ii) = _theFitter->Result().Correlation(ic, ii);
+         covs(ic, ii) = _theFitter->Result().CovMatrix(ic, ii);
       }
+   }
+   res->fillCorrMatrix(globalCC, corrs, covs);
+
+   return res;
+}
 
-    }
-
-    if (isConst) {
-      constPars.addOwned(std::move(var)) ;
-    } else {
-      var->setError(xerr) ;
-      floatPars.addOwned(std::move(var)) ;
-    }
-  }
-
-  res->setConstParList(constPars) ;
-  res->setInitParList(floatPars) ;
-  res->setFinalParList(floatPars) ;
-  res->setMinNLL(_theFitter->Result().MinFcnValue()) ;
-  res->setEDM(_theFitter->Result().Edm()) ;
-  res->setCovQual(_theFitter->GetMinimizer()->CovMatrixStatus()) ;
-  res->setStatus(_theFitter->Result().Status()) ;
-  std::vector<double> globalCC;
-  TMatrixDSym corrs(_theFitter->Result().Parameters().size()) ;
-  TMatrixDSym covs(_theFitter->Result().Parameters().size()) ;
-  for (unsigned int ic=0; ic<_theFitter->Result().Parameters().size(); ic++) {
-    globalCC.push_back(_theFitter->Result().GlobalCC(ic));
-    for (unsigned int ii=0; ii<_theFitter->Result().Parameters().size(); ii++) {
-      corrs(ic,ii) = _theFitter->Result().Correlation(ic,ii);
-      covs(ic,ii) = _theFitter->Result().CovMatrix(ic,ii);
-    }
-  }
-  res->fillCorrMatrix(globalCC,corrs,covs) ;
-
-  return res;
-
-}
-
-void RooMinimizer::setEvalErrorWall(bool flag) { _fcn->SetEvalErrorWall(flag); }
+void RooMinimizer::setEvalErrorWall(bool flag)
+{
+   _cfg.doEEWall = flag;
+   _fcn->SetEvalErrorWall(_cfg.doEEWall);
+}
 
 /// \copydoc RooMinimizerFcn::SetRecoverFromNaNStrength()
-void RooMinimizer::setRecoverFromNaNStrength(double strength) { _fcn->SetRecoverFromNaNStrength(strength); }
-void RooMinimizer::setPrintEvalErrors(Int_t numEvalErrors) { _fcn->SetPrintEvalErrors(numEvalErrors); }
-void RooMinimizer::setVerbose(bool flag) { _verbose = flag ; _fcn->SetVerbose(flag); }
-bool RooMinimizer::setLogFile(const char* logf) { return _fcn->SetLogFile(logf); }
+void RooMinimizer::setRecoverFromNaNStrength(double strength)
+{
+   _cfg.recoverFromNaN = strength;
+   _fcn->SetRecoverFromNaNStrength(_cfg.recoverFromNaN);
+}
+void RooMinimizer::setPrintEvalErrors(Int_t numEvalErrors)
+{
+   _cfg.printEvalErrors = numEvalErrors;
+   _fcn->SetPrintEvalErrors(_cfg.printEvalErrors);
+}
+void RooMinimizer::setVerbose(bool flag)
+{
+   _cfg.verbose = flag;
+   _fcn->SetVerbose(_cfg.verbose);
+}
```

Comment:
What's the point of passing around this `verbose` flag also in the `_fcn`, if the values are always set together anyway? I think the `RooAbsMinimizerFcn::_verbose` data member should better be removed, and then instead of using `_verbose` in the minimizer functions, you use `_context->_cfg.verbose` (as `_context` is a pointer to the owning RooMinimizer).

What do you think? This would make the emphasis of the configuration struct as the single source of truth stronger. 

The change would also entail getting rid of passing the `verbose` flag to some of the RooAbsMinimizerFcn functions and slim down their interfaces (It made no sense for example that the RooMinimizer passes a verbosity flag to the `_fcn` at construction time, when the `_fcn` had it's own verbosity flag that was always synced anyway).